### PR TITLE
ZIP file uploads

### PR DESCRIPTION
**Pull request by MartinBasti**
Implemented handling of zip file uploads and extraction which will be used by operator courier in later PRs.

Missing parts:

- [x] fix failing tests
- [x] add tests for zip files
- [x] put defaults from Config object to flask config
- [x] documentation in README

### CODE REVIEW

File: omps/push.py [hunk 1]
```diff
@@ -2,12 +2,86 @@
...
+                    "name={zi.filename}, compress_size={zi.compress_size}, "
+                    "file_size={zi.file_size}".format(zi=zipinfo)
+                    for zipinfo in archive.filelist
+                )
+            )
+
+        uncompressed_size = sum(zi.file_size for zi in archive.filelist)
+        if uncompressed_size > max_uncompressed_size:
+            raise OMPSUploadedFileError(
+                "Uncompressed archive may reach max size limit "
```
Comment (athos-ribeiro) [inline omps/push.py]: any reason for not saing the uncompressed archive is larger than the limit?

File: tests/conftest.py [hunk 2]
```diff
@@ -46,3 +50,23 @@ def valid_manifests_archive(datadir, tmpdir):
...
+
+@pytest.fixture(params=[
+    True,  # endpoint with version
+    False,  # endpoint without version
+])
+def endpoint_push_zipfile(request):
+    """Returns URL for zipfile endpoints"""
+    organization = 'organization-X'
+    repo = 'repo-Y'
+    version = '0.0.1' if request.param else None
```
Comment (athos-ribeiro) [inline tests/conftest.py]: Should this 0.0.1 come from the default version constant?

File: tests/push/test_api.py [hunk 3]
```diff
@@ -8,23 +8,24 @@
...
         )
 
     assert rv.status_code == 200, rv.get_json()
     expected = {
-        'organization': 'organization-X',
-        'repo': 'repo-Y',
+        'organization': endpoint_push_zipfile.org,
+        'repo': endpoint_push_zipfile.repo,
         'msg': 'Not Implemented. Testing only',
+        'version': endpoint_push_zipfile.version or '0.0.1',
```
Comment (athos-ribeiro) [inline tests/push/test_api.py]: Should this 0.0.1 come from the default version constant?

File: tests/conftest.py (hunk 2 above)
Comment (MartinBasti) [reply tests/conftest.py]: No here, I decided to keep property different from defaults (default value will change)

File: tests/push/test_api.py (hunk 3 above)
Comment (MartinBasti) [reply tests/push/test_api.py]: yes, new default will be also "1.0.0"

**Comment by MartinBasti** [comment]
I hope I've addressed all comments as expected

**Pull request merged**